packages/astro/src/content/utils.ts

```diff
@@ -358,3 +387,90 @@ function search(fs: typeof fsMod, srcDir: URL) {
 	}
 	return { exists: false, url: paths[0] };
 }
+
+export async function getStringifiedLookupMap({
+	contentPaths,
+	contentEntryConfigByExt,
+	root,
+	fs,
+}: {
+	contentEntryConfigByExt: ReturnType<typeof getContentEntryConfigByExtMap>;
+	contentPaths: Pick<ContentPaths, 'contentDir' | 'cacheDir'>;
+	root: URL;
+	fs: typeof fsMod;
+}) {
+	const { contentDir } = contentPaths;
+	const globOpts: FastGlobOptions = {
+		absolute: true,
+		cwd: fileURLToPath(root),
+		fs: {
+			readdir: fs.readdir.bind(fs),
+			readdirSync: fs.readdirSync.bind(fs),
+		},
+	};
+
+	const relContentDir = rootRelativePath(root, contentDir, false);
+	const contentGlob = await glob(
+		`${relContentDir}**/*${getExtGlob([...contentEntryConfigByExt.keys()])}`,
+		globOpts
+	);
+	let filePathByLookupId: {
+		[collection: string]: Record<string, string>;
+	} = {};
+
+	await Promise.all(
+		contentGlob.map(async (filePath) => {
+			const info = getEntryInfo({ contentDir, entry: filePath });
+			if (info instanceof NoCollectionError) return;
```

Comment:
Is there a reason why we are silencing an error?

Reply:
Ah, this shouldn't be an error in hindsight. Just means the entry isn't inside a collection directory. I'm refactoring this implementation to something more readable in the data collections PR. Will add a clarifying comment here.